src/fides/api/ctl/routes/system.py

```diff
@@ -0,0 +1,211 @@
+from loguru import logger
+from pydantic.types import conlist
+from sqlalchemy.orm import Session
+from starlette.status import (
+    HTTP_200_OK,
+    HTTP_400_BAD_REQUEST,
+    HTTP_404_NOT_FOUND,
+    HTTP_422_UNPROCESSABLE_ENTITY,
+)
+
+from fides.api.ctl.sql_models import System  # type: ignore[attr-defined]
+from fides.api.ctl.utils.api_router import APIRouter
+from fides.api.ops.api import deps
+from fides.api.ops.api.v1.endpoints.connection_endpoints import (
+    requeue_requires_input_requests,
+    validate_secrets,
+)
+from fides.api.ops.api.v1.scope_registry import (
+    CONNECTION_CREATE_OR_UPDATE,
+    CONNECTION_READ,
+)
+from fides.api.ops.api.v1.urn_registry import (
+    CONNECTION_TYPES,
+    SYSTEM_CONNECTIONS,
+    V1_URL_PREFIX,
+)
+from fides.api.ops.models.connectionconfig import ConnectionConfig
+from fides.api.ops.schemas.api import BulkUpdateFailed
+from fides.api.ops.schemas.connection_configuration.connection_config import (
+    BulkPutConnectionConfiguration,
+    ConnectionConfigurationResponse,
+    CreateConnectionConfigurationWithSecrets,
+    SaasConnectionTemplateValues,
+)
+from fides.api.ops.service.connectors.saas.connector_registry_service import (
+    create_connection_config_from_template_no_save,
+    load_registry,
+    registry_file,
+)
+from fides.api.ops.util.oauth_util import verify_oauth_client
+from fides.lib.exceptions import KeyOrNameAlreadyExists
+
+router = APIRouter(tags=["System"], prefix=f"{V1_URL_PREFIX}{SYSTEM_CONNECTIONS}")
+
+
+def validate_system(db: Session, fides_key: str) -> System:
+    system = System.get_by(db, field="fides_key", value=fides_key)
+    if system is None:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail="A valid system must be provided to create or update connections",
+        )
+    return system
+
+
+@router.get(
+    "",
+    dependencies=[Security(verify_oauth_client, scopes=[CONNECTION_READ])],
+    status_code=HTTP_200_OK,
+    response_model=Page[ConnectionConfigurationResponse],
+)
+def get_system_connections(
+    fides_key: str, params: Params = Depends(), db: Session = Depends(deps.get_db)
+) -> AbstractPage[ConnectionConfigurationResponse]:
+    """
+    Return all the connection configs related to a system.
+    """
+    system = validate_system(db, fides_key)
+    query = ConnectionConfig.query(db)
+    query = query.filter(ConnectionConfig.system_id == system.id)
+    return paginate(query.order_by(ConnectionConfig.name.asc()), params=params)
+
+
+@router.patch(
+    "",
+    dependencies=[Security(verify_oauth_client, scopes=[CONNECTION_CREATE_OR_UPDATE])],
+    status_code=HTTP_200_OK,
+    response_model=BulkPutConnectionConfiguration,
+)
+def patch_connections(
+    fides_key: str,
+    configs: conlist(CreateConnectionConfigurationWithSecrets, max_items=50),  # type: ignore
+    db: Session = Depends(deps.get_db),
+) -> BulkPutConnectionConfiguration:
```

Comment:
I guess I also don't understand the reasoning behind creating a separate route here - could the `system_id` be added as an optional param to the other route? Or is it that the other route will eventually be deprecated?

Reply:
Good question. It could have been implemented that way. The main reason I implemented it this way is the UI is going to be updated to only allow connections to be created if they're associated with a system. So orphaned systems will go away.

I decided to do it this way so that would be enforced from the backend. Forming the route this way made sense to me at the time because that would match the flow they were describing. I modeled it after the connection/dataset routes. For example, `/connection/{connection_key}/dataset`. `/system/{system_fides_key}/connection` follows that.

I think it would make sense to deprecate the original method once the UI has been updated. That wasn't talked about though.

Reply:
If we were to support both ways long term then the query param would have been better. I'm _pretty_ sure were moving solely to this new way so I made this new route instead

Reply:
ok that's fine